xds/src/test/java/io/grpc/xds/LookasideChannelLbTest.java

```diff
@@ -0,0 +1,433 @@
+        .build();
+    LbEndpoint endpoint22 = LbEndpoint.newBuilder()
+        .setEndpoint(Endpoint.newBuilder()
+            .setAddress(Address.newBuilder()
+                .setSocketAddress(SocketAddress.newBuilder()
+                    .setAddress("addr22").setPortValue(22))))
+        .setLoadBalancingWeight(UInt32Value.of(22))
+        .build();
+    Locality localityProto3 = Locality.newBuilder()
+        .setRegion("region3").setZone("zone3").setSubZone("subzone3").build();
+    LbEndpoint endpoint3 = LbEndpoint.newBuilder()
+        .setEndpoint(Endpoint.newBuilder()
+            .setAddress(Address.newBuilder()
+                .setSocketAddress(SocketAddress.newBuilder()
+                    .setAddress("addr31").setPortValue(31))))
+        .setLoadBalancingWeight(UInt32Value.of(31))
+        .build();
+    ClusterLoadAssignment clusterLoadAssignment = ClusterLoadAssignment.newBuilder()
+            .addEndpoints(LocalityLbEndpoints.newBuilder()
+                .setLocality(localityProto1)
+                .addLbEndpoints(endpoint11)
+                .addLbEndpoints(endpoint12)
+                .setLoadBalancingWeight(UInt32Value.of(1)))
+            .addEndpoints(LocalityLbEndpoints.newBuilder()
+                .setLocality(localityProto2)
+                .addLbEndpoints(endpoint21)
+                .addLbEndpoints(endpoint22)
+                .setLoadBalancingWeight(UInt32Value.of(2)))
+            .addEndpoints(LocalityLbEndpoints.newBuilder()
+                .setLocality(localityProto3)
+                .addLbEndpoints(endpoint3)
+                .setLoadBalancingWeight(UInt32Value.of(0)))
+            .build();
+    serverResponseWriter.onNext(
+        DiscoveryResponse.newBuilder()
+            .addResources(Any.pack(clusterLoadAssignment))
+            .setTypeUrl("type.googleapis.com/envoy.api.v2.ClusterLoadAssignment")
+            .build());
+
+    XdsLocality locality1 = XdsLocality.fromLocalityProto(localityProto1);
+    LocalityInfo localityInfo1 = new LocalityInfo(
+        ImmutableList.of(
+            new XdsComms.LbEndpoint(endpoint11),
+            new XdsComms.LbEndpoint(endpoint12)),
+        1, 0);
+    LocalityInfo localityInfo2 = new LocalityInfo(
+        ImmutableList.of(
+            new XdsComms.LbEndpoint(endpoint21),
+            new XdsComms.LbEndpoint(endpoint22)),
+        2, 0);
+    XdsLocality locality2 = XdsLocality.fromLocalityProto(localityProto2);
+
+    InOrder inOrder = inOrder(localityStore);
+    inOrder.verify(localityStore).updateDropPercentage(ImmutableList.<DropOverload>of());
+    inOrder.verify(localityStore).updateLocalityStore(localityEndpointsMappingCaptor.capture());
+    assertThat(localityEndpointsMappingCaptor.getValue()).containsExactly(
+        locality1, localityInfo1, locality2, localityInfo2).inOrder();
+
+    verify(adsStreamCallback, never()).onError();
+
+    lookasideChannelLb.shutdown();
+  }
+
+  @Test
+  public void onAdsStreamError() {
```

Comment:
May want to rename to `rpcErrorPropagateToLoadBalancer`.

Reply:
Rename to `verifyRpcErrorPropagation`